src/main/java/seedu/address/model/attendance/AttendanceList.java

```diff
@@ -0,0 +1,106 @@
+package seedu.address.model.attendance;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+
+import javafx.collections.transformation.FilteredList;
+import seedu.address.model.person.NusNetId;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Student;
+import seedu.address.model.person.UniqueStudentsInTutorialList;
+
+/**
+ * Contains a list of all students attendance lists.
+ * The students are in the same Tutorial.
+ */
+public class AttendanceList {
+    private final ArrayList<Attendance> attendances;
+    private final int weeks;
+
+    /**
+     * Constructs an AttendanceList.
+     *
+     * @param attendances the list of attendance records.
+     * @param weeks the number of weeks to for attendance to be recorded for.
+     */
+    public AttendanceList(ArrayList<Attendance> attendances, int weeks) {
+        requireAllNonNull(attendances);
+        this.attendances = attendances;
+        this.weeks = weeks;
+    }
+
+    public int getWeeks() {
+        return weeks;
+    }
+
+    public ArrayList<Attendance> getAttendances() {
+        return attendances;
+    }
+
+    /**
+     * Generates attendance records for students who do not have any records present in the list.
+     *
+     * @param uniqueStudentList the list of students in the tutorial.
+     */
+    public void generateAttendance(UniqueStudentsInTutorialList uniqueStudentList) {
+        FilteredList<Person> studentList = uniqueStudentList.getStudentsInClass();
+        for (Person p : studentList) {
+            Student student = (Student) p;
+            NusNetId studentId = student.getStudentId();
+            boolean hasAttendencePresent = false;
+
+
+            for (Attendance attendance : attendances) {
+                if (attendance.getStudentId().equals(studentId)) {
+                    hasAttendencePresent = true;
+                    break;
+                }
+            }
+
+            if (!hasAttendencePresent) {
+                System.out.println("creating");
+                System.out.println(weeks);
```

Comment:
For now will you still need this line to check whether you method works?

Reply:
Thanks for pointing it out. I'll remove it!